evoseal/core/events.py

```
@@ -0,0 +1,368 @@
+            print(f"Workflow started: {event.data}")
+
+        # As a direct call
+        def on_step_completed(event: Event) -> None:
+            print(f"Step completed: {event.data}")
+
+        subscribe(EventType.STEP_COMPLETED, on_step_completed)
+    """
+    return event_bus.subscribe(*args, **kwargs)
+
+
+def publish(event: Union[Event, str], **kwargs: Any) -> Event:
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)